### PR TITLE
test(signer): Add test with duplicate signer private key

### LINKED ISSUE #5077: Add an integration test for duplicate signers on the same network
We should have a test that spins up multiple signers with the same private key and ensure the system can handle signatures from duplicate signers. The first submitted valid signature should take precedence.

### PR DESCRIPTION
**Pull request by jbencin**
### Description

- Add v0 signer test where first two signers have the same key
- Add checks in `mine_and_verify_confirmed_naka_block()` to confirm that no two signers have the same private key

### Applicable issues

- fixes #5077

### Additional info (benefits, drawbacks, caveats)

I was trying to make this test check that only the first signer with the duplicate was accepted, and the second was rejected, but this seems impossible because the signatures from the signers are identical. So I just check that the signature is not included twice

### Checklist

- [x] Test coverage for new or modified code paths
- [x] New integration test(s) added to `bitcoin-tests.yml`

### CODE REVIEW

File: testnet/stacks-node/src/tests/signer/v0.rs [hunk 1]
```diff
@@ -2945,3 +2953,59 @@ fn min_gap_between_blocks() {
...
+        None,
+        Some(signer_stacks_private_keys),
+    );
+
+    signer_test.boot_to_epoch_3();
+    let timeout = Duration::from_secs(30);
+
+    info!("------------------------- Try mining one block -------------------------");
+
+    signer_test.mine_and_verify_confirmed_naka_block(timeout, num_signers);
```
Comment (jferrant) [inline testnet/stacks-node/src/tests/signer/v0.rs:3008]: I would add a check here with something along these lines.
```rust
        let chunks = test_observer::get_stackerdb_chunks();
        let mut found_signatures = vec![];
        for chunk in chunks
            .into_iter()
            .map(|chunk| chunk.modified_slots)
            .flatten()
        {
            let Ok(SignerMessage::BlockResponse(BlockResponse::Accepted((hash, signature)))) = SignerMessage::consensus_deserialize(&mut chunk.data.as_slice())
            else {
                continue;
            };
            if duplicate_pubkey.verify(&hash, signature).expect("failed to verify signature") {
                found_signatures.push(signature);
            }
        }
        assert_eq!(found_signatures.len(), 2);
        assert_eq!(found_signatures[0], found_signatures[1]);
   ```

Comment (jbencin) [reply testnet/stacks-node/src/tests/signer/v0.rs:3008]: I'm working on it now. The thing is there are a lot of messages and they don't seem to be in a specific order, so it's going to be more complicated than this

Comment (jbencin) [reply testnet/stacks-node/src/tests/signer/v0.rs:3009]: Added checks in [bf860f4](https://github.com/stacks-network/stacks-core/pull/5108/commits/bf860f4d0fc41c34abc418632d778cefcc8fc2c3)

**Review by jferrant — APPROVED** [review]

**Review by obycode — APPROVED** [review]
LGTM!
